utils/swift_build_support/swift_build_support/products/swiftpm.py

```diff
@@ -126,8 +145,13 @@ def has_cross_compile_hosts(self, args):
     def get_install_destdir(self, args, host_target, build_dir):
         install_destdir = args.install_destdir
         if self.has_cross_compile_hosts(args):
-            build_root = os.path.dirname(build_dir)
-            install_destdir = '%s/intermediate-install/%s' % (build_root, host_target)
+            if StdlibDeploymentTarget.get_target_for_name(
+                    host_target).platform.is_darwin:
+                build_root = os.path.dirname(build_dir)
+                install_destdir = '%s/intermediate-install/%s' % (
+                                  build_root, host_target)
+            else:
+                install_destdir = os.path.join(install_destdir, host_target)
```

Comment:
@drexin, I just noticed that [you added a method after I submitted this pull that fully replicates this logic from `build-script-impl`](https://github.com/apple/swift/pull/37608/files#diff-b84b1e74590d3a1234977cf25386482b020e325198c52f47625938c940a49e89R204), rather than this partially-recreated version here. What do you think of dropping this method and using your new method in its place, once this pull is in to allow cross-compiling these later `build-script` products too?

Since you're familiar with this build logic, could you review this pull?

Reply:
I've added a new commit with those changes, can push that into a separate pull if you want.

Upon looking into this, there was a subtle difference on Darwin that I had not noticed before: [`build-script-impl` products for macOS that are LIPO'd are installed in the build directory](https://github.com/apple/swift/blob/8198d82845f84f0c75bd4356fd37f49e7929a33e/utils/build-script-impl#L1096), which the above logic reproduces, but [your new method places them in `install_destdir`](https://github.com/apple/swift/pull/37608/files#diff-b84b1e74590d3a1234977cf25386482b020e325198c52f47625938c940a49e89R211). I fixed that in the new commit, let me know if that's right.